### PR TITLE
Spa tabs

### LINKED ISSUE #75: Load tab-specific content only after clicking on tab
When on the site-details page, we should only load / request data needed for the overview page.

After clicking a tab we should load content for that tab & un-load previous tab.
This will reduce requests, speed things up, and make things up to date while navigating the site instead of needing a refresh. It will also more closely resemble a SPA, which we are trying to mimic

Same goes for dashboard/account settings!

**While doing this we really should split up those files a bit. site-details.js is a horrendous 2500+ lines long**

### PR DESCRIPTION
**Pull request by DriftingSands**
Refactor of site-details and dashboard to reduce code repetition and split things into their own files to make updates easier

Fix #75 

Test URLs:
- Before: https://main--eds-self-service-website--headwire-edge-delivery.aem.live/dashboard
- After: https://spa-tabs--eds-self-service-website--headwire-edge-delivery.aem.live/dashboard

### CODE REVIEW

**Comment by DriftingSands** [comment]
Action run with updated tests: https://github.com/headwire-edge-delivery/self-service-tests/actions/runs/11275890816

Failed due to autoTour interference.

**Comment by DriftingSands** [comment]
Test action URL: https://github.com/headwire-edge-delivery/self-service-tests/actions/runs/11277867360

File: head.html [hunk 1]
```diff
@@ -5,7 +5,7 @@
 <link rel="stylesheet" href="/styles/styles.css"/>
 <script src="https://cdn.auth0.com/js/auth0-spa-js/2.0/auth0-spa-js.production.js"></script>
 <script src="/scripts/auth0.js" type="module"></script>
-<script defer src='https://static.cloudflareinsights.com/beacon.min.js' data-cf-beacon='{"token": "7ef9e67e566e49269a85c45427652359"}'></script>
-<script src="/cdn-cgi/zaraz/i.js" referrerpolicy="origin"></script>
+<!-- <script defer src='https://static.cloudflareinsights.com/beacon.min.js' data-cf-beacon='{"token": "7ef9e67e566e49269a85c45427652359"}'></script>
```
Comment (icaraps) [inline head.html:8]: Why did you comment this out ?

**Review by icaraps — CHANGES_REQUESTED** [review]

File: head.html (hunk 1 above)
Comment (DriftingSands) [reply head.html:8]: my bad. was annoyed by the console errors of them on localhost, couldnt find my logs

Comment (icaraps) [reply head.html:8]: maybe we can add the script conditionally if its on kestrelone.com

**Pull request merged**